spec/listeners/automation_rule_listener_spec.rb

```diff
@@ -584,6 +588,41 @@
     end
   end
 
+  describe '#conversation_updated to to detect current language' do
+    before do
+      automation_rule.update!(
+        event_name: 'conversation_updated',
+        name: 'resolve conversation if the language is not english',
+        description: 'Conversation resolved',
+        conditions: [{ 'values': ['en'], 'attribute_key': 'current_language', 'query_operator': nil, 'filter_operator': 'not_equal_to' }],
+        actions: [{ 'action_name' => 'resolve_conversation' }.with_indifferent_access]
+      )
+    end
+
+    let(:client) { double }
+    let(:message) { create(:message, account: account, conversation: conversation, message_type: 'incoming', content: 'muchas muchas gracias') }
+
+    context 'when rule matches' do
+      it 'triggers automation rule to report spam and resolves the conversation' do
+        perform_enqueued_jobs do
+          create(:integrations_hook, :google_translate, account_id: account.id)
+          response = Google::Cloud::Translate::V3::DetectLanguageResponse.new({ languages: [{ language_code: 'es', confidence: 0.71875 }] })
+
+          allow(Google::Cloud::Translate).to receive(:translation_service).and_return(client)
+          allow(client).to receive(:detect_language).and_return(response)
+          message
+        end
```

Comment:
We don't need to process the `translate` related jobs here? We can test them separately in a test for `detect_language_job` .  In this case lets create the conversation object with `conversation_language` populated.

Reply:
Oh with `conversation_language` attribute then we don't need `require 'google/cloud/translate/v3'` here.